code/game/objects/items/weapons/tools/upgrades.dm

```diff
@@ -0,0 +1,445 @@
+/*
+	A tool upgrade is a little attachment for a tool that improves it in some way
+	Tool upgrades are generally permanant
+
+	Some upgrades have multiple bonuses. Some have drawbacks in addition to boosts
+*/
+
+/*/client/verb/debugupgrades()
+	for (var/t in subtypesof(/obj/item/weapon/tool_upgrade))
+		new t(usr.loc)
+*/
+
+/obj/item/weapon/tool_upgrade
+	name = "tool upgrade"
+	icon = 'icons/obj/tool_upgrades.dmi'
+	force = WEAPON_FORCE_HARMLESS
+	w_class = ITEM_SIZE_SMALL
+
+	var/prefix = "upgraded" //Added to the tool's name
+
+	//The upgrade can be applied to a tool that has any of these qualities
+	var/list/required_qualities = list()
+
+	//If true, can only be applied to tools that use fuel
+	var/req_fuel = FALSE
+
+	//If true, can only be applied to tools that use a power cell
+	var/req_cell = FALSE
+
+	var/obj/item/weapon/tool/holder = null //The tool we're installed into
+	matter = list(MATERIAL_STEEL = 1)
+
+	//Actual effects of upgrades
+	var/precision = 0
+	var/workspeed = 0
+	var/degradation_mult = 1
+	var/force_mult = 1	//Multiplies weapon damage
+	var/force_mod = 0	//Adds a flat value to weapon damage
+	var/powercost_mult = 1
+	var/fuelcost_mult = 1
+	var/bulk_mod = 0
+
+/obj/item/weapon/tool_upgrade/examine(var/mob/user)
+	.=..()
+	if (precision > 0)
+		user << SPAN_NOTICE("Enhances precision by [precision]")
+	else if (precision < 0)
+		user << SPAN_WARNING("Reduces precision by [abs(precision)]")
+	if (workspeed)
+		user << SPAN_NOTICE("Enhances workspeed by [workspeed*100]%")
+
+	if (degradation_mult < 1)
+		user << SPAN_NOTICE("Reduces tool degradation by [(1-degradation_mult)*100]%")
+	else if	(degradation_mult > 1)
+		user << SPAN_WARNING("Increases tool degradation by [(degradation_mult-1)*100]%")
+
+	if (force_mult != 1)
+		user << SPAN_NOTICE("Increases tool damage by [(force_mult-1)*100]%")
+	if (force_mod)
+		user << SPAN_NOTICE("Increases tool damage by [force_mod]")
+	if (powercost_mult != 1)
+		user << SPAN_WARNING("Modifies power usage by [(powercost_mult-1)*100]%")
+	if (fuelcost_mult != 1)
+		user << SPAN_WARNING("Modifies fuel usage by [(fuelcost_mult-1)*100]%")
+	if (bulk_mod)
+		user << SPAN_WARNING("Increases tool size by [bulk_mod]")
+
+	if (required_qualities.len)
+		user << SPAN_WARNING("Requires a tool with one of the following qualities:")
+		user << english_list(required_qualities, and_text = " or ")
+
+/******************************
+	UPGRADE TYPES
```

Comment:
Upgrades themselves should be putted into new file. Should be easier to work with them.